### ISSUE TITLE
[tRPC] CI breaks everytime there's a new version of Angular (even minor)

### ISSUE THREAD

**Issue by mparpaillon** [analogjs/analog#1461]
### Please provide the environment you discovered this bug in.

"@analogjs/content": "1.9.2",
"@analogjs/router": "1.9.2",
"@analogjs/trpc": "0.2.46",
"@angular/animations": "18.2.12",
"@angular/cdk": "18.2.12",
"@angular/common": "18.2.12",
"@angular/compiler": "18.2.12",
"@angular/core": "18.2.12",
"@angular/forms": "18.2.12",
"@angular/material": "18.2.12",
"@angular/platform-browser": "18.2.12",
"@angular/platform-browser-dynamic": "18.2.12",
"@angular/platform-server": "18.2.12",
"@angular/router": "18.2.12",

### Which area/package is the issue in?

trpc

### Description

I feel like something is wrong with the peerDependencies in @analogjs/trpc because everytime there's a new version of Angular, my CI breaks and force me to update to the latest Angular version.

I tried to force the angular version in my package.json and it didn't fix it. 

Maybe am I doing something wrong?

### Please provide the exception or error you saw

```true
Build failed with status: FAILURE and message: npm error code ERESOLVE
  npm error ERESOLVE unable to resolve dependency tree
  npm error
  npm error While resolving: @optee/source-prod@0.0.0
  npm error Found: @angular/core@18.2.12
  npm error node_modules/@angular/core
  npm error   @angular/core@"18.2.12" from the root project
  npm error   peer @angular/core@"^16.0.0 || ^17.0.0 || ^18.0.0" from @analogjs/trpc@0.2.46
  npm error   node_modules/@analogjs/trpc
  npm error     @analogjs/trpc@"0.2.46" from the root project
  npm error
  npm error Could not resolve dependency:
  npm error peer @angular/core@"18.2.13" from @angular/common@18.2.13
  npm error node_modules/@angular/common
  npm error   peer @angular/common@"^16.0.0 || ^17.0.0 || ^18.0.0" from @analogjs/trpc@0.2.46
  npm error   node_modules/@analogjs/trpc
  npm error     @analogjs/trpc@"0.2.46" from the root project
  npm error
  npm error Fix the upstream dependency conflict, or retry
  npm error this command with --force or --legacy-peer-deps
  npm error to accept an incorrect (and potentially broken) dependency resolution.
  npm error
  npm error
  npm error For a full report see:
  npm error /www-data-home/.npm/_logs/2024-11-27T08_03_15_658Z-eresolve-report.txt
  npm error A complete log of this run can be found in: /www-data-home/.npm/_logs/2024-11-27T08_03_15_658Z-debug-0.log. For more details see the logs at https://console.cloud.google.com/cloud-build/builds;region=europe-west1/d2d039f4-bee9-4868-a376-7350b0e3d437?project=22449048409.
```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by brandonroberts** [comment]
This happens with every major due to the peer dependencies issue. I'll bump the release of `@analogjs/trpc` to include v19

**Comment by brandonroberts** [comment]
Fixed in `@analogjs/trpc` version `0.2.47`

**Comment by brandonroberts** [comment]
Interesting. As far as semver is concerned, `18.2.12` should be covered under `^18.0.0`

**Comment by mparpaillon** [comment]
I agree. Super weird

**Comment by mparpaillon** [comment]
Hi @brandonroberts I've found the issue. It seems to be caused by the Firebase preset that creates a package.json with hardcoded semver like this

```json
{
  "name": "@optee/source-prod",
  "version": "0.0.0",
  "type": "module",
  "private": true,
  "dependencies": {
    "@analogjs/trpc": "0.2.47",
    "@angular/core": "19.0.5",
    "@faker-js/faker": "8.4.1",
    "@fastify/busboy": "3.1.0",
    "@supabase/auth-js": "2.61.0",
    "@supabase/functions-js": "2.4.4",
    "@supabase/node-fetch": "2.6.15",
    "@supabase/postgrest-js": "1.17.7",
    "@supabase/realtime-js": "2.11.2",
    "@supabase/storage-js": "2.7.1",
    "@supabase/supabase-js": "2.47.10",
    "@trpc/client": "10.45.2",
    "@trpc/server": "10.45.2",
    "abort-controller": "3.0.0",
    "agent-base": "6.0.2",
    "agentkeepalive": "4.5.0",
    "base64-js": "1.5.1",
    "bignumber.js": "9.1.2",
    "buffer-equal-constant-time": "1.0.1",
    "call-bind-apply-helpers": "1.0.1",
    "call-bound": "1.0.3",
    "cookie-es": "1.2.2",
    "copy-anything": "3.0.5",
    "cors": "2.8.5",
    "debug": "4.4.0",
    "defu": "6.1.4",
    "destr": "2.0.3",
    "drizzle-orm": "0.32.2",
    "dunder-proto": "1.0.1",
    "ecdsa-sig-formatter": "1.0.11",
    "encoding": "0.1.13",
    "es-define-property": "1.0.1",
    "es-errors": "1.3.0",
    "es-object-atoms": "1.0.0",
    "event-target-shim": "5.0.1",
    "extend": "3.0.2",
    "firebase-admin": "13.0.2",
    "firebase-functions": "6.2.0",
    "form-data-encoder": "1.7.2",
    "formdata-node": "4.4.1",
    "function-bind": "1.1.2",
    "gaxios": "5.1.3",
    "gcp-metadata": "6.1.0",
    "get-intrinsic": "1.2.6",
    "google-auth-library": "9.15.0",
    "gopd": "1.2.0",
    "gtoken": "7.1.0",
    "h3": "1.13.0",
    "has-flag": "4.0.0",
    "has-symbols": "1.1.0",
    "hasown": "2.0.2",
    "https-proxy-agent": "5.0.1",
    "humanize-ms": "1.2.1",
    "iconv-lite": "0.6.3",
    "iron-webcrypto": "1.2.1",
    "is-stream": "2.0.1",
    "is-what": "4.1.16",
    "isomorphic-fetch": "3.0.0",
    "jose": "4.15.9",
    "json-bigint": "1.0.0",
    "jsonwebtoken": "9.0.2",
    "jwa": "2.0.0",
    "jwks-rsa": "3.1.0",
    "jws": "4.0.0",
    "limiter": "1.1.5",
    "lodash.clonedeep": "4.5.0",
    "lodash.includes": "4.3.0",
    "lodash.isboolean": "3.0.3",
    "lodash.isinteger": "4.0.4",
    "lodash.isnumber": "3.0.3",
    "lodash.isplainobject": "4.0.6",
    "lodash.isstring": "4.0.1",
    "lodash.once": "4.1.1",
    "lru-cache": "6.0.0",
    "lru-memoizer": "2.3.0",
    "mailersend": "2.3.0",
    "math-intrinsics": "1.1.0",
    "ms": "2.1.3",
    "node-domexception": "1.0.0",
    "node-fetch": "2.7.0",
    "node-forge": "1.3.1",
    "object-assign": "4.1.1",
    "object-inspect": "1.13.3",
    "ohash": "1.1.4",
    "openai": "4.77.0",
    "postgres": "3.4.5",
    "qs": "6.13.1",
    "radix3": "1.1.2",
    "rxjs": "7.8.1",
    "safe-buffer": "5.2.1",
    "safer-buffer": "2.1.2",
    "semver": "7.6.3",
    "side-channel": "1.1.0",
    "side-channel-list": "1.0.0",
    "side-channel-map": "1.0.1",
    "side-channel-weakmap": "1.0.2",
    "superjson": "2.2.2",
    "supports-color": "7.2.0",
    "tr46": "0.0.3",
    "tslib": "2.8.1",
    "ufo": "1.5.4",
    "uncrypto": "0.1.3",
    "unenv": "1.10.0",
    "uuid": "9.0.1",
    "vary": "1.1.2",
    "web-streams-polyfill": "4.0.0-beta.3",
    "webidl-conversions": "3.0.1",
    "whatwg-url": "5.0.0",
    "ws": "8.18.0",
    "yallist": "4.0.0",
    "zod": "3.24.1"
  },
  "main": "index.mjs",
  "engines": {
    "node": "18"
  }
}
```

Angular just released 19.0.6 so the broke again because of this.

![Capture d’écran 2025-01-09 à 11 31 33](https://github.com/user-attachments/assets/e4278b47-f8c3-43b5-a3bb-a25c44853aa0)

Are those dependencies really necessary since the build has already created the bundle before deploying to Firebase ?

**Comment by brandonroberts** [comment]
Yea, the `@angular/core` and `@analogjs/trpc` dependencies shouldn't be in that `package.json` as they are bundled in the app. Do you have any extra config in your `vite.config.ts`?

You can try adding `@analogjs/trpc` and `@angular/core` to the `ssr.noExternal` array in the vite config and check to see if they still show up in the package.json.

**Comment by brandonroberts** [comment]
Also, why is it trying to install using that package.json? Everything should be present in the build output directory including `node_modules`

**Comment by mparpaillon** [comment]
Those are good questions.
I'm not using ssr so I don't use noExternal either

Here's why vite.config.ts

```ts
/// <reference types="vitest" />

import analog from "@analogjs/platform";
import path from "path";
import { defineConfig } from "vite";
import Checker from "vite-plugin-checker";
import tsconfigPaths from "vite-tsconfig-paths";
import environment from "../../getEnv";

export default defineConfig(({ mode }) => {
  // Source: https://vitejs.dev/config/#using-environment-variables-in-config

  const extraPlugins =
    mode === "production"
      ? []
      : [
          Checker({
            typescript: {
              tsconfigPath: path.resolve(__dirname, "./tsconfig.app.json"),
            },
          }),
        ];

  return {
    root: __dirname,
    cacheDir: `../../node_modules/.vite`,

    optimizeDeps: {
      include: [
        "@angular/common",
        "@angular/core",
        "@angular/forms",
        "@angular/animations",
        "@angular/cdk/dialog",
        "@angular/cdk/portal",
        "@angular/cdk/overlay",
        "@angular/cdk/table",
        "drizzle-orm/pg-core",
        "@supabase/supabase-js",
        "@typebot.io/js/web",
        "@trpc/client",
        "@trpc/server",
        "zod",
        "rxjs-zone-less",
        "primeng/tooltip",
        "primeng/paginator",
        "primeng/checkbox",
        "primeng/selectbutton",
        "primeng/select",
        "primeng/multiselect",
        "primeng/textarea",
        "primeng/button",
        "primeng/datepicker",
        "primeng/dropdown",
        "primeng/stepper",
        "ngxtension/host-binding",
      ],
    },

    build: {
      outDir: "../../dist/apps/marketplace/client",
      reportCompressedSize: true,
      sourcemap: true,
      target: ["es2020"],
    },
    server: {
      host: true,
      fs: {
        allow: [
          "."
        ],
      },
    },
    plugins: [
      analog({
        ssr: false,
        vite: {
          inlineStylesExtension: "scss",
        },
        nitro: {
          preset: "firebase",
          firebase: {
            nodeVersion: "18",
            gen: 2,
            serverFunctionName: environment.FIREBASE_FN_NAME,
            httpsOptions: {
              region: "europe-west1",
              maxInstances: 10,
              cpu: 1,
              memory: "2GiB",
              concurrency: 80,
              minInstances: environment.FIREBASE_FN_NAME === "server" ? 1 : 0,
            },
          },
        },
      }),

      tsconfigPaths(),
      ...extraPlugins,
    ],
    define: {
      "import.meta.vitest": mode !== "production",
    },
  };
});

```

It's honestly quite a black box for me. I don't know what the firebase preset actually does or how it works

**Comment by brandonroberts** [comment]
Ahh ok. Nitro builds the API server and produces the `package.json` in the `server` folder in the output directory. You shouldn't need to run `npm install` on that package.json which is the part that's confusing. 

The reason those deps are included in the package.json is likely because `@analogjs/trpc` imports from `@angular/core` and is only used in the server build. If you search through the js files in the `server` folder, do you see any imports from `@angular/core`?

**Comment by brandonroberts** [comment]
You can inline the `@analogjs/trpc` and `@angular/core` dependencies by adding this to the `nitro` object. Then they are removed from that package.json in the server directory.

```ts
          externals: {
            inline: ['@angular/core', '@analogjs/trpc']
          }
```

It will make the base server bundle larger though

**Comment by mparpaillon** [comment]
Thanks Brandon. I'm not the one running the npm install, it seems to be a default firebase behavior on deploy. I don't know if there's a way to prevent it. It's taking CI time for nothing. I'm giving "externals" a try on the nitro config. Thanks

**Comment by mparpaillon** [comment]
>  If you search through the js files in the server folder, do you see any imports from @angular/core

Sorry forgot to answer that question. No there is no import from @angular/core in the server folder. 

Also I tried the fix above and it broke the install. At first the nx reset took 30+ minutes and I had to cancel it (don't know why) and after the build would simply not work because of some memory issue. 

It seems that Firebase automatically runs npm install on deploy so this package.json is a problem.

Angular 19.1.0 just dropped and my CI just broke.

![Capture d’écran 2025-01-16 à 10 38 19](https://github.com/user-attachments/assets/3dbc34ef-471d-402a-bb29-540fe81b88a5)
